v2/internal/testcommon/kube_test_context_envtest.go

```diff
@@ -102,14 +104,19 @@ func createSharedEnvTest(cfg testConfig, namespaceResources *namespaceResources)
 
 	var cacheFunc cache.NewCacheFunc
 	if cfg.TargetNamespaces != nil && cfg.OperatorMode.IncludesWatchers() {
-		cacheFunc = cache.MultiNamespacedCacheBuilder(cfg.TargetNamespaces)
+		cacheFunc = func(config *rest.Config, opts cache.Options) (cache.Cache, error) {
```

Comment:
Consider extracting this out in a method and call it here directly?

Reply:
I did consider that. It's passed as a callback that's invoked later and needs to capture locals from the current context. I'm not sure the refactored method would be any clearer.

Reply:
I'll check out the possibility of a refactor offline; want to get this merged to clear dependabot PRs.